var/spack/repos/builtin/packages/vigra/package.py

```diff
@@ -0,0 +1,85 @@
+    """VIGRA stands for "Vision with Generic Algorithms". It's an image
+       processing and analysis library that puts its main emphasis on
+       customizable algorithms and data structures"""
+
+    homepage = "https://ukoethe.github.io/vigra/"
+    url = "https://github.com/ukoethe/vigra/releases/download/Version-1-11-1/vigra-1.11.1-src.tar.gz"
+
+    version('1.11.1', sha256='a5564e1083f6af6a885431c1ee718bad77d11f117198b277557f8558fa461aaf')
+
+    variant('png', default=False, description='Include support for PNG images')
+    variant('jpeg', default=False, description='Include support for JPEG images')
+    variant('tiff', default=False, description='Include support for TIFF images')
+    variant('exr', default=False, description='Include support for EXR images')
+    variant('hdf5', default=False, description='Include support for HDF5 files')
+    variant('fftw', default=False, description='Include support for Fourier Transforms')
+    variant('python', default=False, description='Generate Python bindings and doc')
+    variant('cxxdoc', default=False, description='Generate C++ documentation')
+
+    depends_on('libtiff', when='+tiff')
+    depends_on('libpng', when='+png')
+    depends_on('libjpeg', when='+jpeg')
+    depends_on('hdf5', when='+hdf5')
+    depends_on('fftw', when='+fftw')
+    depends_on('openexr', when='+exr')
+    depends_on('py-numpy', when='+python')
+    depends_on('boost+python+numpy', when='+python')
+    depends_on('py-sphinx', when='+python')
+    depends_on('doxygen', when='+cxxdoc')
+    depends_on('python', when='+cxxdoc')
+    depends_on('py-nose', type=('build', 'test'), when='+python')
+
+    extends('python')
+
+    def cmake_args(self):
+        args = []
+        spec = self.spec
+
+        if '+tiff' in spec:
+            args.extend([
+                '-DTIFF_LIBRARY={0}'.format(spec['libtiff'].libs),
+                '-DTIFF_INCLUDE_DIR={0}'.format(spec['libtiff'].prefix.include),
+            ])
+        if '+hdf5' in spec:
+            args.extend([
+                '-DHDF5_CORE_LIBRARY={0}'.format(spec['hdf5'].libs.libraries[0]),
+                '-DHDF5_INCLUDE_DIR={0}'.format(spec['hdf5'].prefix.include),
+                '-DHDF5_Z_LIBRARY={0}'.format(spec['zlib'].libs.libraries[0])
+            ])
+        if '+python' in spec:
+            pyVersStr = spec['python'].version.up_to(2)
+            pyVersStrNoDot = pyVersStr.joined
+            pyVersDir = 'python{0}'.format(pyVersStr)
+            boostPythonLib = '{0}/libboost_python{1}.so'.format(
+                    spec['boost'].prefix.lib, pyVersStrNoDot)
+            args.extend([
+                '-DBoost_DIR={0}'.format(spec['boost'].prefix),
+                '-DBoost_INCLUDE_DIR={0}'.format(spec['boost'].prefix.include),
+                '-DBoost_PYTHON_LIBRARY={0}'.format(boostPythonLib),
+                '-DVIGRANUMPY_INSTALL_DIR={0}'.format(join_path(spec.prefix.lib,
+                        pyVersDir, 'site-packages'))
```

Comment:
Brilliant. Did not know about this one. In my experience (pre-Spack), many packages try to install Python bindings into the Python dir instead of with the package, so good to know that Spack has a generic mechanism for this. I think I could have used this at least once before, so I should go back and check.